src_main/xevdm.c

```diff
@@ -2793,13 +2800,14 @@ int xevd_dec_nalu(XEVD_CTX * ctx, XEVD_BITB * bitb, XEVD_STAT * stat)
         msh->alf_on = sps->tool_alf;
 
         msh->mmvd_group_enable_flag = sps->tool_mmvd;
-        ctx->sps_id++;
+        //ctx->sps_id++; no
     }
     else if (nalu->nal_unit_type_plus1 - 1 == XEVD_NUT_PPS)
     {
         ret = xevdm_eco_pps(bs, sps, pps);
         xevd_assert_rv(XEVD_SUCCEEDED(ret), ret);
-        ctx->sps = &ctx->sps_array[pps->pps_seq_parameter_set_id];
+        ctx->sps_id = pps->pps_seq_parameter_set_id % 16;
+        sps = ctx->sps = &ctx->sps_array[ ctx->sps_id ];
```

Comment:
sps_id should not be based on pps_seq_parameter_set_id. Please check.

Reply:
yes, ctx->sps_count is unused in the current xevd code. 
So, then there is no need to check ctx->sps_id is out of bounds. Also ctx->sps_id is based on sps->sps_seq_parameter_set_id in current xevd code.